apps/mobile/src/components/GalleryEditor/GalleryEditorRow.tsx

```diff
@@ -42,50 +52,45 @@ export function GalleryEditorRow({ sectionId, row, style, queryRef }: Props) {
     [activateRow, sectionId, row.id]
   );
 
+  const handleDragStart = useCallback(() => {
+    activateRow(sectionId, row.id);
+  }, [activateRow, sectionId, row.id]);
+
+  const handleDragEnd = useCallback(
+    (newPositionsById: string[]) => {
+      moveItem(row.id, newPositionsById);
+    },
+    [moveItem, row.id]
+  );
+
   return (
-    <Animated.View className={clsx('border border-transparent gap-4')}>
+    <Animated.View className={clsx('border border-transparent')}>
       <GalleryTouchableOpacity
         eventElementId={null}
         eventName={null}
         eventContext={null}
         onPress={handleSectionPress}
-        className={clsx('border border-transparent relative', {
-          'border-activeBlue': activeRowId === row.id,
-        })}
-        style={style}
+        withoutFeedback
       >
-        <View>
-          <View className="flex-row flex-wrap gap-2">
-            {row.items.map((item) => {
-              if (item.kind === 'whitespace') {
-                return <WhiteSpace key={item.id} size={widthPerToken - 8} />;
-              } else {
-                return (
-                  <View
-                    key={item.id}
-                    className="aspect-square"
-                    style={{
-                      width: widthPerToken - 8,
-                    }}
-                  >
-                    <GalleryEditorTokenPreview tokenRef={item.tokenRef} />
-                  </View>
-                );
-              }
-            })}
+        <View
+          className={clsx('border border-transparent relative', {
+            'border-activeBlue': activeRowId === row.id,
+          })}
+        >
+          <View className="relative">
+            <SortableTokenGrid
+              columns={row.columns}
+              items={row.items}
+              size={widthPerToken}
+              scrollContentOffsetY={scrollContentOffsetY}
+              scrollViewRef={scrollViewRef}
+              onDragStart={handleDragStart}
+              onDragEnd={handleDragEnd}
+            />
```

Comment:
This component responsible to **render** the item in grid view

Reply:
can we add this as a comment?